tests/command/test_sync.rs

```diff
@@ -97,3 +99,71 @@ fn test_sync_up_to_date() -> eyre::Result<()> {
     }
     Ok(())
 }
+
+#[test]
+fn test_sync_pull() -> eyre::Result<()> {
+    let GitWrapperWithRemoteRepo {
+        temp_dir: _guard,
+        original_repo,
+        cloned_repo,
+    } = make_git_with_remote_repo()?;
+
+    original_repo.init_repo()?;
+    original_repo.commit_file("test1", 1)?;
+    original_repo.commit_file("test2", 2)?;
+    original_repo.run(&["checkout", "-b", "foo"])?;
+    original_repo.commit_file("test3", 3)?;
+
+    original_repo.clone_repo_into(&cloned_repo, &["--branch", "master"])?;
+    cloned_repo.init_repo_with_options(&GitInitOptions {
+        make_initial_commit: false,
+        ..Default::default()
+    })?;
+    cloned_repo.run(&["checkout", "foo"])?;
+
+    original_repo.commit_file("test4", 4)?;
+    original_repo.run(&["checkout", "master"])?;
+    original_repo.commit_file("test5", 5)?;
+
+    {
+        let (stdout, _stderr) = cloned_repo.run(&["smartlog"])?;
+        insta::assert_snapshot!(stdout, @r###"
+        :
+        O 96d1c37a (master, remote origin/master) create test2.txt
+        |
+        @ 70deb1e2 (> foo) create test3.txt
+        "###);
+    }
+
+    {
+        let (stdout, _stderr) = cloned_repo.run(&["sync", "-p"])?;
+        insta::assert_snapshot!(stdout, @r###"
+        branchless: running command: <git-executable> fetch --all
+        Fetching origin
+        Attempting rebase in-memory...
+        [1/1] Committed as: 8e521a10 create test3.txt
+        branchless: processing 1 update: branch foo
+        branchless: processing 1 rewritten commit
+        branchless: running command: <git-executable> checkout foo
```

Comment:
Sorry about an off-topic comment for this PR, but these message have always seemed to me like they should be at debug level. Or do you think users care about them (I've only used git-branchless as few times, so I don't count myself as a user)?

Reply:
In this particular case, since it's delegating to a sub-operation, a lot of the messages are grayed out:

<img width="694" alt="Zrzut ekranu 2022-03-13 o 01 22 36" src="https://user-images.githubusercontent.com/454057/158053218-56feee89-5cf2-49d8-a021-11dd8017f676.png">

For the `branchless:` messages, the reasoning was originally:

1) to help debug in early stages of development.
2) to inform users that git-branchless reacts to/tracks user events.

My intention was to add an option to disable these messages for users who understood how git-branchless works, but in practice, I never got around to it because it didn't bother me. (I've been careful to output only a constant number of lines per operation, unlike Git, which might output 100 lines after a commit with 100 files.)

Reply:
Makes sense. Thanks for explaining.